app/controllers/claims/support/schools_controller.rb

```diff
@@ -6,4 +6,44 @@ def index
   def show
     @school = Claims::School.find(params.require(:id))
   end
+
+  def new
+    @school = Claims::School.new
+  end
+
+  def check
+    if school.valid? && !school.claims?
+      @school = school.decorate
+    else
+      school.errors.add(:urn, :taken) if school.claims?
+      render :new
+    end
+  end
+
+  def create
+    school.claims = true
+    if school.save
+      redirect_to claims_support_schools_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def school
+    @school ||=
+      begin
+        gias_school = GiasSchool.find_by(urn: urn_param)
+        if gias_school.blank?
+          Claims::School.new
+        else
+          gias_school.school || gias_school.build_school
+        end
+      end
+  end
```

Comment:
Out of curiosity, why do we _start_ with the `GiasSchool` model? Could we do something along the lines of

```suggestion
  def school
    @school ||= Claims::School.find_or_initialize_by(gias_school:)
  end
  
  def gias_school
    @gias_school ||= GiasSchool.find_by(urn: urn_param) 
  end
```

This would ensure we always have the namespaced `School` model, right?

Reply:
We can separate out the `GiasSchool` out if you really want to. But again, the `Claim::School` won't find schools which already exist as `Placements::School`. Therefore this will error when trying to save this record, as the `:urn` will have already been taken and a school record already exists.

Reply:
Ah yes, we want to share the data. I overlooked that line of thought. Makes sense to me.

Reply:
Feel free to ignore this suggestion :)

Reply:
Also, that implementation won't account for when the `GiasSchool` is `nil`